### PR TITLE
Fea: Voice muting

### PR DESCRIPTION
**Pull request by mzegar**

### CODE REVIEW

File: code/Game.cs [hunk 1]
```diff
@@ -104,13 +104,19 @@ public override void RenderHud()
...
 			return false;
 
-		return true;
+		if ( !sourcePlayer.IsAlive() && destPlayer.CanHearSpectators )
+			return true;
+
+		if ( State is InProgress && sourcePlayer.IsAlive() && destPlayer.CanHearAlivePlayers )
+			return true;
+
+		return false;
```
Comment (mzegar) [inline code/Game.cs:119]: Let me double check this logic quickly...

**Comment by mzegar** [comment]
image

File: code/Game.ConCmds.cs [hunk 2]
```diff
@@ -87,6 +87,26 @@ public static void ToggleForceSpectator()
...
+		player.PlayersMuted = player.PlayersMuted switch
+		{
+			PlayersMute.None => PlayersMute.AlivePlayers,
+			PlayersMute.AlivePlayers => PlayersMute.Spectators,
+			PlayersMute.Spectators => PlayersMute.All,
+			PlayersMute.All => PlayersMute.None,
+			_ => PlayersMute.None
+		};
+	}
+
```
Comment (kolexxx) [inline code/Game.ConCmds.cs:109]: Why is this serverside?

File: code/Player/Player.Spectating.cs [hunk 3]
```diff
@@ -2,11 +2,25 @@
...
+	All
+}
+
 public partial class Player
 {
 	[Net, Local]
 	public bool IsForcedSpectator { get; private set; } = false;
 
+	[Net, Local]
+	public PlayersMute PlayersMuted { get; set; } = PlayersMute.None;
```
Comment (kolexxx) [inline code/Player/Player.Spectating.cs:19]: This needs to be replaced with `ConVar.ClientData`

File: code/Player/Player.Spectating.cs [hunk 4]
```diff
@@ -113,4 +127,11 @@ private static void OnPlayerKilled( Player player )
 		if ( localPlayer.IsSpectatingPlayer && localPlayer.CurrentPlayer == player )
 			localPlayer.Camera = new FreeSpectateCamera();
 	}
+
+	[GameEvent.Round.Start]
+	private void UnmutePlayers()
+	{
+		if ( Host.IsServer )
+			PlayersMuted = PlayersMute.None;
+	}
```
Comment (kolexxx) [inline code/Player/Player.Spectating.cs:136]: Should be clientside.

**Review by kolexxx — CHANGES_REQUESTED** [review]
Let's create a `Player.Communication` file

**Comment by mzegar** [comment]
Closes https://github.com/sbox-TTT/TTT/issues/163

**Review by kolexxx — APPROVED** [review]

**Pull request merged**